Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+def androidAPILevel = '23'
+
+def sdkSetup(sdkVersion) {
+  sh 'npm install titanium'
+  sh './node_modules/.bin/ti login travisci@appcelerator.com travisci'
+  echo "Installing ${sdkVersion}"
+  sh "./node_modules/.bin/ti sdk install -d ${sdkVersion} --no-progress-bars"
+  def sdkListJSON = jsonParse(sh(returnStdout: true, script: './node_modules/.bin/ti sdk list -o json'))
+  def titaniumRoot = sdkListJSON['defaultInstallLocation']
+  def activeSDKVersion = sdkListJSON['activeSDK']
+  return sdkListJSON['installed'][activeSDKVersion]
+}
+
+def buildAndroid() {
+  return {
+    node('android-sdk && android-ndk && (osx || linux)') {
+      unstash 'sources'
+      nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+        def activeSDKPath = sdkSetup(tiSDKVersion)
+
+        // We have to hack to make sure we pick up correct ANDROID_SDK/NDK values from the node that's currently running this section of the build.
+        withEnv(['ANDROID_SDK=', 'ANDROID_NDK=']) {
+          def androidSDK = sh(returnStdout: true, script: 'printenv ANDROID_SDK')
+          def androidNDK = sh(returnStdout: true, script: 'printenv ANDROID_NDK')
+          sh "./node_modules/.bin/ti config android.sdkPath ${androidSDK}"
+          sh "./node_modules/.bin/ti config android.ndkPath ${androidNDK}"
+          dir('android') {
+            writeFile file: 'build.properties', text: """
+titanium.platform=${activeSDKPath}/android
+android.platform=${androidSDK}/platforms/android-${androidAPILevel}
+google.apis=${androidSDK}/add-ons/addon-google_apis-google-${androidAPILevel}
+"""
+            // TODO Use 'appc ti build --build-only'!
+            // if lib folder doesn't exist, create it
+            sh 'mkdir -p lib'
+            // if build folder doesn't exist, create it
+            sh 'mkdir -p build'
+            // if build/docs folder doesn't exist, create it
+            sh 'mkdir -p build/docs'
+            def antHome = tool(name: 'Ant 1.9.2', type: 'ant')
+            withEnv(["PATH+ANT=${antHome}/bin"]) {
+              sh 'ant clean'
+              sh 'ant'
+            }
+            archiveArtifacts 'dist/*.zip'
+          } // dir
+        } // withEnv
+      } // nodeJs
+    } // node
+  }
+}
+
+def buildIOS() {
+  return {
+    node('osx && xcode && python') {
+      unstash 'sources'
+      nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+        def activeSDKPath = sdkSetup(tiSDKVersion)
+
+        dir('ios') {
+          writeFile file: 'titanium.xcconfig', text: """
+TITANIUM_SDK = ${activeSDKPath}
+TITANIUM_BASE_SDK = \"\$(TITANIUM_SDK)/iphone/include\"
+TITANIUM_BASE_SDK2 = \"\$(TITANIUM_SDK)/iphone/include/TiCore\"
+TITANIUM_BASE_SDK3 = \"\$(TITANIUM_SDK)/iphone/include/ASI\"
+TITANIUM_BASE_SDK4 = \"\$(TITANIUM_SDK)/iphone/include/APSHTTPClient\"
+HEADER_SEARCH_PATHS= \$(TITANIUM_BASE_SDK) \$(TITANIUM_BASE_SDK2) \$(TITANIUM_BASE_SDK3) \$(TITANIUM_BASE_SDK4) \${PROJECT_DIR}/**
+"""
+          sh './build.py'
```

Comment:
We should use `ti build -p ios --build-only` or the Appc-CLI `appc run -p ios --build-only`.